drift-transport-netty/src/test/java/com/facebook/drift/transport/netty/codec/TestHeaderTransport.java

```diff
@@ -21,69 +21,141 @@
 import com.facebook.drift.protocol.TProtocolWriter;
 import com.facebook.drift.transport.netty.buffer.TestingPooledByteBufAllocator;
 import com.facebook.drift.transport.netty.ssl.TChannelBufferOutputTransport;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufAllocator;
 import org.testng.annotations.Test;
 
+import java.util.List;
 import java.util.Optional;
 
 import static com.facebook.drift.protocol.TMessageType.CALL;
 import static com.facebook.drift.protocol.TMessageType.ONEWAY;
+import static com.facebook.drift.transport.netty.codec.HeaderTransport.decodeFrame;
+import static com.facebook.drift.transport.netty.codec.HeaderTransport.encodeFrame;
 import static com.facebook.drift.transport.netty.codec.HeaderTransport.tryDecodeFrameInfo;
 import static com.facebook.drift.transport.netty.codec.Protocol.BINARY;
 import static com.facebook.drift.transport.netty.codec.Protocol.FB_COMPACT;
+import static com.facebook.drift.transport.netty.codec.ThriftHeaderTransform.ZLIB_TRANSFORM;
 import static com.facebook.drift.transport.netty.codec.Transport.HEADER;
 import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
 
 public class TestHeaderTransport
 {
+    private static final int FULL_LENGTH = -1;
+
+    @Test
+    public void testNoTransform()
+            throws Exception
+    {
+        assertTryDecodeSequenceId(ImmutableList.of(), 0, Optional.empty());
+        assertTryDecodeSequenceId(ImmutableList.of(), 1, Optional.empty());
+        assertTryDecodeSequenceId(ImmutableList.of(), 5, Optional.empty());
+        assertTryDecodeSequenceId(ImmutableList.of(), 10, Optional.empty());
+        assertTryDecodeSequenceId(ImmutableList.of(), 15, Optional.empty());
+        assertTryDecodeSequenceId(ImmutableList.of(), 50,
+                Optional.of(new FrameInfo("method", CALL, 0xFFAA, HEADER, BINARY, true)));
+        assertTryDecodeSequenceId(ImmutableList.of(), FULL_LENGTH,
+                Optional.of(new FrameInfo("method", CALL, 0xFFAA, HEADER, BINARY, true)));
+        assertRoundTrip(ImmutableList.of());
+    }
+
     @Test
-    public void testTryDecodeSequenceId()
+    public void testZlibTransform()
+            throws Exception
+    {
+        assertTryDecodeSequenceId(ImmutableList.of(ZLIB_TRANSFORM), 0, Optional.empty());
+        assertTryDecodeSequenceId(ImmutableList.of(ZLIB_TRANSFORM), 1, Optional.empty());
+        assertTryDecodeSequenceId(ImmutableList.of(ZLIB_TRANSFORM), 5, Optional.empty());
+        assertTryDecodeSequenceId(ImmutableList.of(ZLIB_TRANSFORM), 10, Optional.empty());
+        assertTryDecodeSequenceId(ImmutableList.of(ZLIB_TRANSFORM), 15, Optional.empty());
+        assertTryDecodeSequenceId(ImmutableList.of(ZLIB_TRANSFORM), 50, Optional.empty());
```

Comment:
Add `assertTryDecodeSequenceId(ImmutableList.of(ZLIB_TRANSFORM), 100, Optional.of(...));`